crates/rattler_repodata_gateway/src/gateway/error.rs

```diff
@@ -34,6 +35,15 @@ pub enum GatewayError {
 
     #[error("the operation was cancelled")]
     Cancelled,
+
+    #[error("the direct url query failed for {0}")]
+    DirectUrlQueryError(String, #[source] DirectUrlQueryError),
+
+    #[error("the match spec '{0}' does not specify a name")]
+    MatchSpecWithoutName(MatchSpec),
+
+    #[error("the package from url '{0}', doesn't have the same name as the match spec filename intents '{1}'")]
+    NotMatchingNameUrl(String, String),
```

Comment:
```suggestion
    UrlRecordNameMismatch(String, String),
```

Reply:
fixed in https://github.com/mamba-org/rattler/pull/725/commits/7ea414a9b3ecf609d1c7027ad99a9e99c50e232